src/lib/ctable.lua

```diff
@@ -217,6 +217,30 @@ function CTable:insert(hash, key, value, updates_allowed)
    return index
 end
 
+-- Choose a random index between the start of the table and its occupancy_hi
+-- value. This guarantees that there will be an entry to eject, because this
+-- is only called when the table is 'full'.
+local function random_eject(ctab)
+   local eject_index = math.random(0, ctab.occupancy_hi - 1)
```

Comment:
This is the wrong way to do things, I think.  Assume that there are 400 slots in the table and that occupancy_high is 350.  Those up-to-350 entries will be randomly distributed across slots 0 to 400+max_displacement.  To find an entry to displace then we should start at some index in [0,400+max_displacement) and search forward, wrapping around if needed.  The strategy you have above will avoid evicting the (size + max_displacement - occupancy_high)/(size+max_displacement) entries at the end, which is probably not what we want -- the expectation would then be that over time this end of the table would be overpopulated with stale entries.